contracts/Controller.sol

```diff
@@ -91,8 +98,16 @@ contract Controller is Ownable {
      * @dev can only be called when system is not paused
      * @param _actions array of actions arguments
      */
-    // function operate(Actions.ActionArgs[] memory _actions) external isNotPaused {
-    // }
+    function operate(Actions.ActionArgs[] memory _actions) external isNotPaused nonReentrant {
+        MarginAccount.Vault memory vault = _runActions(_actions);
+
+        address calculatorModule = AddressBookInterface(addressBook).getMarginCalculator();
+        MarginCalculatorInterface calculator = MarginCalculatorInterface(calculatorModule);
+
+        if (vault.shortOtokens.length > 0) {
+            calculator.isValidState(vault, vault.shortOtokens[0]);
+        }
```

Comment:
Not sure about this, if we should only call `isValidState` when there is a short otoken in vault.
But can't pass an invalid index `vault.shortOtokens[0]` when  there is no short otoken.

Reply:
This doesn't sounds good for me, why the calculator need `address(0)` even if there is no short otoken ?

Reply:
yeah right now that parameter is really redundant, I feel like it would make everyone's life easier if we can remove it for now. Our original idea was to keep the interface consistent for future calculator updates, not sure if we still a requirement to hold.

Reply:
Also `isValidState()` return bool right? so I should probably add a require statement there.